src/com/amazon/ion/impl/IonWriterSystemText.java

```diff
@@ -151,14 +146,14 @@ void push(int typeid)
         _stack_pending_comma[_top] = _pending_separator;
         switch (typeid) {
         case _Private_IonConstants.tidSexp:
-            _separator_character = ' ';
+            _separator_character = " ";
             break;
         case _Private_IonConstants.tidList:
         case _Private_IonConstants.tidStruct:
-            _separator_character = ',';
+            _separator_character = ",";
             break;
         default:
-            _separator_character = _options.isPrettyPrintOn() ? '\n' : ' ';
+            _separator_character = _options.lineSeparator();
```

Comment:
FYI—the logic here is unchanged, except for the fact that it will now use the configured line separator instead of being hard-coded to `\n`. This is one of places where there was an inconsistency in the type of newlines being written by the existing code, but as far as I can tell, this code should be unreachable.

Same thing also applies to the similar line below.